app/screens/find_channels/filtered_list/filtered_list.tsx

```diff
@@ -274,25 +294,40 @@ const FilteredList = ({
             items.push(...archivedAlpha.slice(0, MAX_RESULTS + 1));
         }
 
-        return [...new Set(items)].slice(0, MAX_RESULTS + 1);
-    }, [archivedChannels, channelsMatchStart, channelsMatch, remoteChannels, usersMatch, usersMatchStart, locale, teammateDisplayNameSetting]);
+        const results = [...new Set(items)].slice(0, MAX_RESULTS + 1);
+
+        // Add threads item to show it on the top of the list
+        if (showThreadItem) {
+            results.unshift('thread');
+        }
+
+        return results;
+    }, [archivedChannels, channelsMatchStart, channelsMatch, remoteChannels, usersMatch, usersMatchStart, locale, teammateDisplayNameSetting, showThreadItem]);
 
     useEffect(() => {
         mounted.current = true;
         return () => {
             mounted.current = false;
         };
-    });
+    }, []);
 
     useEffect(() => {
         bounce.current = debounce(search, 500);
         bounce.current();
+        if (isCRTEnabled) {
+            const label = formatMessage({
```

Comment:
you could memoize this label with a dependency on `locale` just to avoid this calculation with every term change